Userland/Libraries/LibGL/Tex/TextureUnit.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <LibGL/Tex/Texture2D.h>
+
+namespace GL {
+
+class TextureUnit {
+public:
+    TextureUnit() = default;
+
+    void bind_texture_to_target(GLenum texture_target, const RefPtr<Texture>& texture);
+    void unbind_texture(GLenum texture_target);
```

Comment:
```suggestion
    void bind(const RefPtr<Texture>& texture);
    void unbind(GLenum texture_target);
```
~~Since only one texture can be bound to a unit~~ and the type of the texture can be queried on the `Texture` class we don't really need to tell the unit which target we want to bind to. This would also prevent us calling `bind_texture_to_target` with conflicting target/texture-type.

EDIT: Screw that. Apparently OpenGL allows you to bind different targets on one texture unit at the same time. *jackie_chan_meme.jpg* https://community.khronos.org/t/binding-different-targets-to-same-unit/76935

Reply:
Yeah this one is a little bit cooked... Not entirely sure how to structure it now. Might be a good think to sit on and have a further read into (I guess we only really support `GL_TEXTURE_2D` for the moment anyway..)

Reply:
According to the spec a target of a texture unit is never really in an unbound state... if you unbind a target with `glBindTexture(GL_TEXTURE_2D, 0)` what you're doing is binding the "default texture"™️ to that target. But yeah, we will iterate here anyway 👍